packages/firestore/src/platform/platform.ts

```diff
@@ -15,69 +15,64 @@
  * limitations under the License.
  */
 
+import { isNode } from '@firebase/util';
+
 import { DatabaseId, DatabaseInfo } from '../core/database_info';
 import { Connection } from '../remote/connection';
 import { JsonProtoSerializer } from '../remote/serializer';
-import { fail } from '../util/assert';
-import { ConnectivityMonitor } from './../remote/connectivity_monitor';
+import { ConnectivityMonitor } from '../remote/connectivity_monitor';
 
-/**
- * Provides a common interface to load anything platform dependent, e.g.
- * the connection implementation.
- *
- * An implementation of this must be provided at compile time for the platform.
- */
-// TODO: Consider only exposing the APIs of 'document' and 'window' that we
-// use in our client.
-export interface Platform {
-  loadConnection(databaseInfo: DatabaseInfo): Promise<Connection>;
-  newConnectivityMonitor(): ConnectivityMonitor;
-  newSerializer(databaseId: DatabaseId): JsonProtoSerializer;
+import * as node from '../platform_node/node_platform';
+import * as browser from '../platform_browser/browser_platform';
 
-  /** Formats an object as a JSON string, suitable for logging. */
-  formatJSON(value: unknown): string;
+// NOTE: This class is replaced in Firestore release builds by either
+// "node_platform" or "browser_platform". The implementations below are only
+// used by ts-node to run tests.
 
-  /** Converts a Base64 encoded string to a binary string. */
-  atob(encoded: string): string;
+/** Loads the network stack (GRPC or WebChannel). */
+export function loadConnection(
+  databaseInfo: DatabaseInfo
+): Promise<Connection> {
+  return isNode()
```

Comment:
I don't think this would be possible unless we define an interface as an intermediate layer. It did however dawn on me that this file is never used in browser builds. I ended up replacing the `isNode` check with an assert.